### PR TITLE
Catch Exception within DefaultDelivery class

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Certain devices such as Blackberries seem to throw exceptions other than IOException within `DefaultDelivery`. This change catches the unexpected failures and drops the payload, while logging a warning, to prevent bugsnag-android causing crashes.

## Changeset

Added a catch clause

## Tests

Ran existing unit tests, mazerunner locally

## Discussion

This behaviour has not been reproduced locally, and appears to be a device-specific issue. It's possible that this could be some inherent issue with the Blackberry repackaging process.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Comment by kattrali** [comment]
As this is just a fix, it should be against master rather than `next`.

File: sdk/src/main/java/com/bugsnag/android/DefaultDelivery.java [hunk 1]
```diff
@@ -79,6 +79,9 @@ int deliver(String urlString,
             return conn.getResponseCode();
         } catch (IOException exception) {
             throw new DeliveryFailureException("IOException encountered in request", exception);
+        } catch (Exception exception) {
+            Logger.warn("Unexpected error delivering payload", exception);
+            return 400;
```
Comment (kattrali) [inline sdk/src/main/java/com/bugsnag/android/DefaultDelivery.java]: This should be made a constant, to make it more declarative why `400` is being used here.

**Comment by fractalwrench** [comment]
Moved to #361 to change base branch against master
